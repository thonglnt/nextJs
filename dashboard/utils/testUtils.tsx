import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
