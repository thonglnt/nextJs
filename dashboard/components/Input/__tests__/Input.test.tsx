// Utils
import { render, screen, fireEvent } from '@utils/testUtils'

// Components
import InputFields, { IInputFields } from '..'

describe('Input component', () => {
  const mockProps: IInputFields = {
    label: 'Input Label',
    name: 'input',
  }

  const mockErrorProps: IInputFields = {
    label: 'Input Label',
    name: 'input',
    errorMsg: 'error',
  }

  it('should render Input component', () => {
    const { container } = render(<InputFields {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render error message', () => {
    const { container } = render(<InputFields {...mockErrorProps} />)

    expect(container).toHaveTextContent('Input Label')
    expect(container).toHaveTextContent('error')
  })

  it('should close Alert after click button close', () => {
    render(<InputFields {...mockProps} />)

    const textFields = screen.getByRole('textbox')

    fireEvent.change(textFields, {
      target: { value: 'Value' },
    })

    expect(textFields).toHaveValue('Value')
  })
})
