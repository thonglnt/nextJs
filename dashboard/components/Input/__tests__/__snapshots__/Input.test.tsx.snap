// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input component 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1tiijee-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-qmlsft-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="outlined-required"
      id="outlined-required-label"
    >
      Input Label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl css-12z6gyx-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="outlined-required"
        name="input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Input Label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
